diff --git a/ocaml/Makefile.in b/ocaml/Makefile.in
index 5d85844..d631bf8 100644
--- a/ocaml/Makefile.in
+++ b/ocaml/Makefile.in
@@ -177,6 +177,8 @@ export OCAMLFLAGS = -thread -warn-error Aelz -annot $(PROJOCAMLFLAGS) # -for-pac
 # merge everything but descriptions.
 export OCAMLDOCFLAGS = -m avlsoper -sort
 
+BFD_TEMP := $(shell mktemp)
+
 all: dcl ncl
 
 bap_lib_doc: all htdoc
@@ -198,12 +200,12 @@ debug:
 libbfd.idl : bfdarch.idl
 
 bfdarch.idl:
-	echo '#include "config.h"' > /tmp/f.c
-	echo '#include <bfd.h>' >> /tmp/f.c
-	echo 'typedef unsigned long machine_t;' >> /tmp/f.c
-	echo 'const machine_t MACH_i386_i386 = bfd_mach_i386_i386;' >> /tmp/f.c
-	echo 'const machine_t MACH_i386_x86_64 = bfd_mach_x86_64;' >> /tmp/f.c
-	$(CPP) $(CPPFLAGS) -I$(LIBASMIR) /tmp/f.c | awk 'BEGIN { go=0; } /^enum bfd_architecture$$/ { go=1; } \
+	echo '#include "config.h"' > $(BFD_TEMP)
+	echo '#include <bfd.h>' >> $(BFD_TEMP)
+	echo 'typedef unsigned long machine_t;' >> $(BFD_TEMP)
+	echo 'const machine_t MACH_i386_i386 = bfd_mach_i386_i386;' >> $(BFD_TEMP)
+	echo 'const machine_t MACH_i386_x86_64 = bfd_mach_x86_64;' >> $(BFD_TEMP)
+	$(CPP) $(CPPFLAGS) -I$(LIBASMIR) -x c $(BFD_TEMP) | awk 'BEGIN { go=0; } /^enum bfd_architecture$$/ { go=1; } \
 		go && $$0 != "" { print; } \
 		/machine_t/ { print; } \
 		/};$$/ { go=0; }' > $@
