--- Makefile.in	2014-03-12 03:16:28.000000000 -0400
+++ Makefile.in.fixed	2014-08-11 19:06:34.704918202 -0400
@@ -3082,12 +3082,9 @@
 # Symlink capnpc -> capnp.  The capnp binary will behave like the old capnpc
 # binary (i.e. like "capnp compile") when invoked via this symlink.
 #
-# Also attempt to run ldconfig, because otherwise users get confused.  If
-# it fails (e.g. because the platform doesn't have it, or because the
-# user doesn't have root privileges), don't worry about it.
+# Do not attempt to run ldconfig, that is not your job.
 install-exec-hook:
 	ln -sf capnp $(DESTDIR)$(bindir)/capnpc
-	ldconfig < /dev/null > /dev/null 2>&1 || true
 
 uninstall-hook:
 	rm -f $(DESTDIR)$(bindir)/capnpc
